src/ui/controllers/voicesearchcontroller.js

```diff
@@ -0,0 +1,135 @@
+   * @param {Object} config The voiceSearch config from the search bar component
+   */
+  constructor (searchBarContainer, config) {
+    this._state = State.NOT_LISTENING;
+
+    /**
+     * The screen reader text for the "Start Voice Search" button
+     * @type {string}
+     */
+    this._startVoiceSearchText = config.startText || TranslationFlagger.flag({
+      phrase: 'Start Voice Search',
+      context: 'A button to begin listening for a voice search'
+    });
+
+    /**
+     * The screen reader text for the "Stop Voice Search" button
+     * @type {string}
+     */
+    this._stopVoiceSearchText = config.stopText || TranslationFlagger.flag({
+      phrase: 'Stop Voice Search',
+      context: 'A button to stop listening for a voice search'
+    });
+
+    this._searchBarContainer = searchBarContainer;
+    this._voiceSearchElement = DOM.query(searchBarContainer, '.yxt-SearchBar-voiceSearch');
+    this._screenReaderTextElement = DOM.query(searchBarContainer, '.yxt-SearchBar-voiceSearchText');
+    this._updateScreenReaderText();
+
+    /** @type {SVGAnimateElement} */
+    this._micFadeInAnimation = DOM.query(searchBarContainer, '.js-yxt-SearchBar-micFadeIn');
+    /** @type {SVGAnimateElement} */
+    this._micFadeOutAnimation = DOM.query(searchBarContainer, '.js-yxt-SearchBar-micFadeOut');
+    /** @type {SVGAnimateElement[]} */
+    this._dotsFadeInAnimations = DOM.queryAll(searchBarContainer, '.js-yxt-SearchBar-dotsFadeIn');
+    /** @type {SVGAnimateElement[]} */
+    this._dotsFadeOutAnimations = DOM.queryAll(searchBarContainer, '.js-yxt-SearchBar-dotsFadeOut');
+
+    this._micIconWrapper = DOM.query(this._searchBarContainer, '.yxt-SearchBar-micIconWrapper');
+    this._listeningIconWrapper = DOM.query(this._searchBarContainer, '.yxt-SearchBar-listeningIconWrapper');
+
+    this._customMicIconUrl = config.customMicIconUrl;
+    this._customListeningIconUrl = config.customListeningIconUrl;
+
+    if (this._customMicIconUrl) {
+      this._applyCustomMicIconStyling();
+    }
+
+    if (this._customListeningIconUrl) {
+      this._applyCustomListeningIconStyling();
+    }
+  }
+
+  handleIconClick () {
+    if (this._state === State.NOT_LISTENING) {
+      this._state = State.LISTENING;
+    } else if (this._state === State.LISTENING) {
+      this._state = State.NOT_LISTENING;
+    }
+    this._updateUI();
+  }
+
+  _updateUI () {
+    if (this._customMicIconUrl) {
+      this._applyCustomMicIconStyling();
+    } else {
+      this._executeDefaultMicIconAnimations();
+    }
+    if (this._customListeningIconUrl) {
+      this._applyCustomListeningIconStyling();
+    } else {
+      this._executeDefaultListeningIconAnimations();
+    }
+    this._updateScreenReaderText();
+    this._voiceSearchElement.blur();
+  }
+
+  _executeDefaultMicIconAnimations () {
+    if (this._state === State.NOT_LISTENING) {
+      this._micFadeInAnimation.beginElement();
```

Comment:
this is cool! did you have to manually write the SVG to use `<animate>` or was UX able to do that all for us?

Reply:
The SVGs that UX provided had the the animate tags in them. I just had to add a css class and `begin="indefinate"` so that I could start them with Javascript rather than on page load.